### PR TITLE
#324 absolute time zone as “location” 

### PR DESCRIPTION
**Pull request by shokhrukh-epam**

### CODE REVIEW

File: src/App/dictionary/en.json [hunk 1]
```diff
@@ -49,6 +49,10 @@
   "Settings.CancelButton": "Cancel",
   "Settings.SaveButton": "Save",
   "Settings.ButtonTooltip": "Settings",
+  "Settings.DisableTimezone": "Disable Timezone",
```
Comment (ArturSalikhovEPAM) [inline src/App/dictionary/en.json:52]: let's use lowercase letter for "Timezone"

File: src/App/dictionary/en.json [hunk 2]
```diff
@@ -49,6 +49,10 @@
...
+  "Settings.AbbreviationTimezone": "Abbreviation Timezone",
```
Comment (ArturSalikhovEPAM) [inline src/App/dictionary/en.json:53]: let's use here "Abbrv. timezone"

File: src/App/dictionary/en.json [hunk 3]
```diff
@@ -49,6 +49,10 @@
...
+  "Settings.CountryTimezone": "Country Timezone",
+  "Settings.AbbreviationAndCountryTimezone": "Abbreviation and Country Timezone",
```
Comment (ArturSalikhovEPAM) [inline src/App/dictionary/en.json:55]: let's use here "Abbrv. and country timezone"

File: src/App/components/Header/components/SettingsModal/SettingsModal.constants.ts [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
 export const SETTING_VALUE = {
   date: 'date',
   country: 'country',
   auto: 'auto',
-  timezone: 'timezone',
+  ...TIMEZONE,
   ...TIME_FORMAT
 };
 
 export const INPUT_IDS = {
```
Comment (ArturSalikhovEPAM) [inline src/App/components/Header/components/SettingsModal/SettingsModal.constants.ts:11]: Can we delete this constant and use just constants from redux?

File: src/App/components/Header/components/SettingsModal/SettingsModal.tsx [hunk 5]
```diff
@@ -144,6 +154,10 @@ const SettingsModal: React.FC = () => {
           <Divider
             className={clsx({ [style.divider]: true, [style.darkDivider]: theme === THEME.dark })}
           />
+          <TimezoneBlock localSettings={localSettings} handleSetSettings={handleSetSettings} />
+          <Divider
```
Comment (ArturSalikhovEPAM) [inline src/App/components/Header/components/SettingsModal/SettingsModal.tsx:158]: I guess we can create separate component "Divider" that will be in folder "UI" and contain fuctionality with useStyle hook

File: src/App/components/Header/components/SettingsModal/components/TimezoneBlock/TimezoneBlock.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import { TIMEZONE } from '../../../../../../redux/constants';
+
+const TimezoneBlock: React.FC<IBlocksProps> = ({ localSettings, handleSetSettings }) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <RadioGroup value={localSettings.showTimezone} onChange={handleSetSettings}>
+        <div>
+          <Radio id={INPUT_IDS.disableTimezone} value={TIMEZONE.disableTimezone} />
```
Comment (ArturSalikhovEPAM) [inline src/App/components/Header/components/SettingsModal/components/TimezoneBlock/TimezoneBlock.tsx:18]: let's also make separate component "RadioGroup"

File: src/App/redux/constants.ts [hunk 7]
```diff
@@ -18,3 +18,10 @@ export const TIME_FORMAT = {
   H24: 'H24',
   H12: 'H12'
 };
+
+export const TIMEZONE = {
+  disableTimezone: 'disableTimezone',
+  abbreviationTimezone: 'abbreviationTimezone',
+  countryTimezone: 'countryTimezone',
+  abbreviationAndCountryTimezone: 'abbreviationAndCountryTimezone'
+};
```
Comment (ArturSalikhovEPAM) [inline src/App/redux/constants.ts:27]: "disabled"
"abbrv"
"country"
"abbrvConutry"

File: src/App/utils/countryToAbbreviationTimezone.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { timezonesDB } from '../redux/timezonesDB';
+
+export default function countryToAbbreviationTimezone(countryTimezone: string | undefined) {
```
Comment (ArturSalikhovEPAM) [inline src/App/utils/countryToAbbreviationTimezone.ts:3]: let create hook for this

**Review by BakhromSadullaev — COMMENTED** [review]
Screenshot (31)
The label of checkbox abbreviation and country timezone is not UI friendly and I think we should change the text

**Comment by SheratriumEPAM** [comment]
@BakhromSadullaev Pls, resolve conflicts

**Review by SheratriumEPAM — APPROVED** [review]
Looks good

**Pull request merged**